Logging/src/LogMessageProcessorFactory.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Copyright 2022 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Logging;
+
+use Google\Cloud\Logging\LogMessageProcessor\MonologMessageProcessor;
+use Google\Cloud\Logging\LogMessageProcessor\MonologV3MessageProcessor;
+use Monolog\Logger as MonologLogger;
+use RuntimeException;
+
+/**
+ * Returns a LogMessageProcessor that can be used with the currently install Monolog version
+ */
+class LogMessageProcessorFactory
+{
+    /**
+     * @return LogMessageProcessor
+     */
+    public function getLogMessageProcessor()
```

Comment:
Let's rename this to `build` and make this static to follow the same convention we use in [HttpHandlerFactory](https://github.com/googleapis/google-auth-library-php/blob/main/src/HttpHandler/HttpHandlerFactory.php):
```suggestion
    public static function build()
```

Reply:
Done!